WalletWasabi.Fluent/ViewModels/Wallets/WalletRenameViewModel.cs

```diff
@@ -0,0 +1,49 @@
+using ReactiveUI;
+using WalletWasabi.Fluent.Models.Wallets;
+using WalletWasabi.Fluent.Validation;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets;
+
+[NavigationMetaData(Title = "Rename Wallet", NavigationTarget = NavigationTarget.CompactDialogScreen)]
+public partial class WalletRenameViewModel : RoutableViewModel
+{
+	[AutoNotify] private string _newWalletName;
+
+	private WalletRenameViewModel(IWalletModel wallet)
+	{
+		_newWalletName = wallet.Name;
+
+		this.ValidateProperty(
+			x => x.NewWalletName,
+			errors =>
+			{
+				if (wallet.Name == NewWalletName)
+				{
+					return;
+				}
+
+				if (UiContext.WalletRepository.ValidateWalletName(NewWalletName) is { } error)
+				{
+					errors.Add(error.Severity, error.Message);
+				}
+			});
+
+		SetupCancel(enableCancel: true, enableCancelOnEscape: true, enableCancelOnPressed: true);
+		var canRename = this.WhenAnyValue(model => model.NewWalletName, selector: newName => !Equals(newName, wallet.Name) && !Validations.Any);
```

Comment:
I think the `!Equals(newName, wallet.Name)` is not needed. I mean having to same name in the field is fine so they can click the continue button, just do not perform any action (similarly to CoinControl dialog when the coins are the same and they click continue).

Reply:
That improves UX, yeah. Fixed.